code/game/objects/structures/girders.dm

```diff
@@ -161,7 +172,25 @@
 				return ..()
 
 	else
-		return ..()
+		var/damage_to_deal = W.force
+		var/weaken = 0
+		if(reinf_material)
+			weaken += reinf_material.integrity * 1.75 //Since girders don't have a secondary material, buff 'em up a bit.
+		weaken /= 100
+		damage_to_deal -= weaken * 1.5
+		visible_message("<span class='notice'>[user] retracts their [W] and starts winding up a strike...</span>")
+		var/hit_delay = W.w_class * 10 //Heavier weapons take longer to swing, yeah?
+		if(do_after(user, hit_delay))
+			do_attack_animation(src)
+			playsound(src, 'sound/weapons/smash.ogg', 50)
+			if(damage_to_deal > weaken || (damage_to_deal < MIN_DAMAGE_TO_HIT))
```

Comment:
That expression does not make sense, because currently the wall would take damage in these cases:

- If damage_to_deal is larger than weaken
- If damage to deal **is smaller** than MIN_DAMAGE_TO_HIT

It should probs be:
```suggestion
			if(damage_to_deal > weaken && (damage_to_deal > MIN_DAMAGE_TO_HIT))
```